### PR TITLE
Remove duplicate ICU package

### PR DESCRIPTION
**Pull request by adamjstewart**
This PR is a followup to #1828. While investigating comments made by @JavierCVilla, I discovered that Spack has a duplicated package! Both `icu` and `icu4c` were downloading the same tarball. It turns out ICU (International Components for Unicode) provides both a C/C++ interface (ICU4C) and a Java interface (ICU4J). Since the tarball contains the text "icu4c" and since there is a separate "icu4j" package that could be added, I think it's best to remove `icu` and point everything to `icu4c`.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Changes to `changed_files` tell `run-flake8-tests` not to try to parse deleted files.

File: var/spack/repos/builtin/packages/boost/package.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ class Boost(Package):
     variant('graph', default=False,
             description="Build the Boost Graph library")
 
-    depends_on('icu', when='+icu_support')
+    depends_on('icu4c', when='+icu_support')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/boost/package.py]: Any chance you could rename the `icu_support` variant to be consistent with the others in `boost`?  Nothing currently depends on `boost+icu_support`.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/boost/package.py]: You want it called `icu`?

Comment (tgamblin) [inline var/spack/repos/builtin/packages/boost/package.py]: yeah -- the others look like that.  The description is also worded differently.  I'm being picky, I know.

Comment (tgamblin) [inline var/spack/repos/builtin/packages/boost/package.py]: `_support` just seems obvious.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/boost/package.py]: Done.

**Pull request merged**